### DISCUSSION TITLE
GitHub Pages Not Reflecting Updated Code

### DISCUSSION THREAD

**Discussion by chsiny** [community/community#144071] [Projects and Issues] [answered]
Hi everyone,

I’m experiencing an issue with my GitHub Pages deployment, where the deployed site is not reflecting the latest code in my repository. Despite confirming that the code on GitHub is updated, the live site remains unchanged.

### Steps Taken So Far

Here’s what I’ve tried to resolve the issue:

1.	Confirmed Code Update: Checked the main branch (the one set up to deploy on GitHub Pages) to ensure it contains the latest code.
2.	Cleared Browser Cache: Cleared the browser cache to rule out any local caching issues.
3.	Forced Cache Refresh: Added <meta> tags in the index.html to prevent caching:
```html
<meta http-equiv="cache-control" content="no-cache" />
<meta http-equiv="expires" content="0" />
<meta http-equiv="pragma" content="no-cache" />
```
4.	Rebuilt and Redeployed: Rebuilt the project with npm run build and pushed the updated build files to GitHub.

### Observations

•	When I test the site locally, it reflects all the latest changes correctly.
•	After each update and deployment, the live site on GitHub Pages still shows an older version, even when I create new branches to deploy.

### Additional Details

•	Repository URL: [my GitHub repository](https://github.com/chsiny/chsiny.github.io)
•	Branch Used for Deployment: main
•	Build Tool: React (Create React App)
•	Deployment Method: Direct GitHub Pages setup

The updated code:
![Screenshot 2024-11-09 at 10 39 14 pm](https://github.com/user-attachments/assets/bdac1449-1937-4e23-a7a2-042ef46540ba)

The deployed view (it should be © 2024 test Designed and coded by Jerry Chen
![Screenshot 2024-11-09 at 10 39 44 pm](https://github.com/user-attachments/assets/81cdb071-ef03-4666-b496-163ff20747de)

### Request for Help

Has anyone else encountered this issue with GitHub Pages not updating? Are there any specific troubleshooting steps or deployment settings that I might be overlooking? I’d appreciate any guidance or suggestions to resolve this.

Thanks in advance for your help!

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by AyhemKouki** [comment]
1) Check GitHub Pages Deployment Status
Ensure that GitHub Pages is correctly set up to deploy from the main branch. Sometimes, if there's an issue during deployment, it won't reflect the latest changes.

To check:
Go to your repository on GitHub.
Navigate to Settings > Pages.
Confirm that the Source is set to the main branch and the folder is set to / (root) or /docs depending on where your build files are.

2. Force Push to GitHub
Sometimes, GitHub Pages doesn't fully recognize updates, especially if you're only modifying existing files. Try doing a force push:
git add .
git commit -m "Force update deployment"
git push origin main --force

**Reply by chsiny** [reply]
Thank you so much for the suggestions!

I tried checking the GitHub Pages deployment settings and ensuring it’s set to deploy from the main branch with the correct folder. I also did a force push as you recommended. Unfortunately, it’s still not reflecting the latest changes on the GitHub Pages site.

If you have any other ideas, I’d really appreciate it. Thanks again for your help!

**Comment by chsiny** [comment, marked as answer]
Thanks to @AyhemKouki  for your time! I’ve resolved the issue. I noticed that the deployment for the main branch kept getting stuck, even after multiple attempts to rerun all jobs. Since I couldn’t view what was happening in the backend, I tried deploying a dedicated gh-pages branch instead.

Here’s what I did:

1. Set up the gh-pages npm package to automate deploying only the build files to a gh-pages branch.
2. Updated my package.json with a "deploy" script:
```json
"scripts": {
  "predeploy": "npm run build",
  "deploy": "gh-pages -d build"
}
```
3. Now, each time I want to update the live site, I just run npm run deploy. This process ensures that the gh-pages branch is kept in sync with the latest build, and GitHub Pages correctly reflects my updates now.
